colossalai/pipeline/schedule/zero_bubble_pp.py

```diff
@@ -472,19 +496,25 @@ def backward_b_step(
         # calculate bwd b step ; only dx = w*dy;
 
         # Retain the grad on the input_obj.
-        tree_map(retain_grad, input_obj)
+        if input_obj is None:
+            return None
+        else:
+            tree_map(retain_grad, input_obj)
+        input_obj_ = input_obj["hidden_states"]
```

Comment:
remove hardcode

Reply:
> remove hardcode
Fixed in 
https://github.com/hpcaitech/ColossalAI/pull/6065/commits/349272c71fa9d30c404ca29b394d7e79bfcd2fd0;